AntennaCharacteristics.ipynb

```diff
@@ -0,0 +1,692 @@
+    "    size=(td.inf, td.inf, 0),\n",
+    "    freqs=freqs_target,  \n",
+    "    name=\"field\",\n",
+    ")\n",
+    "\n",
+    "# Define observation parameters used for DirectitiviyMonitor\n",
+    "theta = np.linspace(-np.pi, np.pi, 200)\n",
+    "phi = np.linspace(0, np.pi, 100)\n",
+    "\n",
+    "# We define a DirectivityMonitor to obatin radiation characteritics such as directivity, axial ratio and polarized far fields.\n",
+    "monitor_directivity = td.DirectivityMonitor(\n",
+    "    center=[0, 0, 0],\n",
+    "    size=(30*mm, 45*mm, 4*mm), # We introduce a monitor box to enclose the whole structure of interest\n",
+    "    freqs=freqs_target,   \n",
+    "    name=\"radiation\",\n",
+    "    phi=list(phi),\n",
+    "    theta=list(theta),\n",
+    "    far_field_approx=True, # Far-field approximation is suitable for most cases. For more accurate far-field projections, set far_field_approx=False.\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Boundaries\n",
+    "\n",
+    "Perfectly Matched Layers (PMLs) are applied in all directions to simulate an infinitely large domain."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# PML wavelength at 10 GHz\n",
+    "wl_pml = wavelength0\n",
+    "\n",
+    "# Quarter wavelength (at 10 GHz) padding on each side\n",
+    "sim_x = sub_x + wl_pml / 2\n",
+    "sim_y = sub_y + wl_pml / 2\n",
+    "sim_z = sub_z + wl_pml / 2\n",
+    "\n",
+    "# Truncate the computational domain by PMLs\n",
+    "boundary_spec = td.BoundarySpec(\n",
+    "    x=td.Boundary.pml(),\n",
+    "    y=td.Boundary.pml(),\n",
+    "    z=td.Boundary.pml(),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Simulation Object\n",
+    "\n",
+    "Create a simulation object that includes all the modules we have defined. Note that the excitation (source) is not defined here and will be introduced by as a lumped port.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create the simulation object\n",
+    "sim = td.Simulation(\n",
+    "    center=[0, 0, 0],\n",
+    "    size=[sim_x,sim_y,sim_z],\n",
+    "    grid_spec=td.GridSpec.auto(\n",
+    "        min_steps_per_wvl=20, # The largest cell size is set to 20 cells per smallest wavelength.\n",
+    "        wavelength=td.C_0 / freq_stop, # Smallest wavelength to resolve  \n",
+    "    ),\n",
+    "    structures=structures_list,  \n",
+    "    sources=[],  # Sources will be added by TerminalComponentModeler\n",
+    "    monitors=[monitor_directivity, monitor_field],\n",
+    "    run_time= 70*(sub_y/td.C_0),   \n",
+    "    boundary_spec=boundary_spec,\n",
+    "    shutoff=1e-5, # Convergence criterion\n",
+    "    subpixel=True, # Use subpixel smoothing for accuracy\n",
```

Comment:
shutoff and subpixel are using the default values. So no need to define them explicitly here.